src/main/java/com/regalii/regaliator/utils/JSON.java

```diff
@@ -39,7 +40,16 @@ static public String dump(final Object object) {
     static public Object loadToMap(final String json) {
         final Gson gson = new Gson();
 
-        return gson.fromJson(json, Map.class);
+        try {
+            return gson.fromJson(json, Map.class);
+        }
+        catch (IllegalStateException | JsonSyntaxException exception) {
+            System.err.println(
+                    String.format(
+                            "Expected a JSON string that is parseable into an object map. Instead, got: \"%s\".",
+                            json));
+            throw exception;
```

Comment:
From the stacktrace we have, I can say that `IllegalStateException` will never throw here, so we only need to catch `JsonSyntaxException`. 

Also I suggest not to log using stderr, but do it in our application instead. We could rewrite it to something like this

```java
catch (JsonSyntaxException e) {
            throw new JsonSyntaxException("Expected a JSON string but got " + json, e);
}

```

Reply:
I think that makes sense. I like the idea of enriching the error itself rather than introducing new logging.